assets/chezmoi.io/docs/user-guide/templating.md

```diff
@@ -93,22 +93,33 @@ $ chezmoi edit --apply ~/.zshrc
 
 ## Testing templates
 
-Templates can be tested with the `chezmoi execute-template` command which treats
-each of its arguments as a template and executes it. This can be useful for
-testing small fragments of templates, for example:
+Templates can be tested and debugged with the `chezmoi execute-template`
+command which treats each of its arguments as a template and executes it.
+It will then interpret the templates and output the result to standard
+output.
+
+This can be useful for testing small fragments of templates, for example:
 
 ```console
 $ chezmoi execute-template '{{ .chezmoi.hostname }}'
 ```
 
 If there are no arguments, `chezmoi execute-template` will read the template
-from the standard input. This can be useful for testing whole files, for example:
+from the standard input. This can be useful for testing whole files,
+for example:
 
 ```console
 $ chezmoi cd
 $ chezmoi execute-template < dot_zshrc.tmpl
 ```
 
+If this does not work (e.g. with Powershell), you can pipe the contents of
```

Comment:
````markdown
If file redirection does not work (as when using PowerShell), the contents of a file can be piped
into `chezmoi execute-template`:

```console
$ cat foo.txt | chezmoi execute-template
```
````

Reply:
This change was not made. The main thing that I care about is `e.g. with PowerShell`; the canonical formatting is `PowerShell` not `Powershell` and `e.g.` is often overused.

Thinking more about this paragraph, I think it would be better (since PowerShell is being used as the example) to use a PowerShell example:

```powershell
Get-Content -Path input.txt -Raw | chezmoi execute-template
```

I’m sure that @bradenhilton will correct me if I got that invocation wrong.

Reply:
> Thinking more about this paragraph, I think it would be better (since PowerShell is being used as the example) to use a PowerShell example:

I don't agree, I am afraid. `cat` works fine in PowerShell (it is [defined as an alias](https://learn.microsoft.com/en-us/powershell/scripting/learn/shell/using-aliases?view=powershell-5.1#compatibility-aliases-in-windows)), but is not limited to it like `Get-Content`.

Reply:
If `cat` works, then that's fine and it does not need changing. I don't use PowerShell, so I was unaware of that default alias.